### PR TITLE
Update to 2021.10.1

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Update fsspec to 2021.10.1

Actions:
1. Add `python <3.10` to test/requires
2. Rearrange host dependencies alphabetically

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,17 +19,18 @@ build:
 
 requirements:
   host:
-    - python >=3.6
```
Comment (katietz) [inline recipe/meta.yaml:22]: please add a skip for python variant less 3.6 ( skip: True. # [py<36] )

**Review by katietz — CHANGES_REQUESTED** [review]
thanks

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:22]: Done

**Comment by katietz** [comment]
Hmm, the failure on osx-arm64 looks a bit suspicious.  Have you taken a look into it, why it fails?  It seems that some script common fails.

**Comment by skupr-anaconda** [comment]
It's a noarch package so it's a weird that it fails on arm64

**Comment by skupr-anaconda** [comment]
Error on arm64:

```
ImportError: dlopen(/Users/builder/miniconda3/envs/prefect/conda-bld/fsspec_1636116543469/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.9/lib-dynload/_ctypes.cpython-39-darwin.so, 2): Library not loaded: @rpath/libffi.7.dylib

  Referenced from: /Users/builder/miniconda3/envs/prefect/conda-bld/fsspec_1636116543469/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.9/lib-dynload/_ctypes.cpython-39-darwin.so

  Reason: image not found
```

**Comment by skupr-anaconda** [comment]
> Hmm, the failure on osx-arm64 looks a bit suspicious. Have you taken a look into it, why it fails? It seems that some script common fails.

It's a `conda` issue https://github.com/conda/conda/issues/9038

**Comment by varlackc** [comment]
There are similar issues with other packages

**Review by katietz — APPROVED** [review]
thanks

**Pull request merged**